### PR TITLE
Are the types wrong?

### PR DESCRIPTION
**Pull request by lishaduck**
Yes they were!

Fixed by dropping CJS from the builds following #225.

Supersedes/closes #161.

### CODE REVIEW

**Comment by lishaduck** [comment]
CI is failing b/c it doesn't exclude as expected. I posted dist/index.js to a gist: https://gist.github.com/lishaduck/0a0d893fc6b3877b82aa29d17f4266f2

**Comment by lishaduck** [comment]
Oh, you know... that nice `noExternal` thing isn't going to keep #150 solved. NPM'll still try to install it. The published package.json needs to not include it all, not just in the bundle. It's in the bundle either way. I'm going to revert that change.

File: packages/sheriff-constants/package.json [hunk 1]
```diff
@@ -1,36 +1,24 @@
 {
   "name": "@sherifforg/constants",
-  "description": "Sheriff constants variables.",
+  "description": "Sheriff constants & variables.",
```
Comment (AndreaPontrandolfo) [inline packages/sheriff-constants/package.json:3]: i think we can keep this as just "Sheriff constants".

File: packages/sheriff-constants/package.json [hunk 2]
```diff
@@ -1,36 +1,24 @@
...
-    }
+    ".": "./src/index.ts"
   },
   "files": [
-    "dist",
+    "src",
     "CHANGELOG.md"
   ],
   "scripts": {
     "publint": "publint",
```
Comment (AndreaPontrandolfo) [inline packages/sheriff-constants/package.json:15]: actually we can remove publint here too. We never ship this as a npm package.

File: packages/sheriff-constants/package.json [hunk 3]
```diff
@@ -1,36 +1,24 @@
...
-    "clean": "rm -rf .turb dist",
-    "build": "tsup",
-    "are-the-types-wrong": "attw --pack ."
+    "clean": "rm -rf .turbo dist"
```
Comment (AndreaPontrandolfo) [inline packages/sheriff-constants/package.json:16]: we can remove dist form here too

File: packages/sheriff-constants/package.json [hunk 4]
```diff
@@ -1,36 +1,24 @@
...
   },
   "devDependencies": {
-    "@arethetypeswrong/cli": "^0.15.4",
     "@sherifforg/types": "workspace:*",
     "publint": "^0.2.10",
```
Comment (AndreaPontrandolfo) [inline packages/sheriff-constants/package.json:20]: we can remove publint here too

**Review by AndreaPontrandolfo — CHANGES_REQUESTED** [review]

**Comment by AndreaPontrandolfo** [comment]
@lishaduck you know, i'm thinking an even better thing we could do to simplify stuff.
Just like with `constants`, we shouldn't ship the @sherifforg/types as a separate package. We can just re-export it from `eslint-config-sheriff.`
I don't see any downsides to it, only upsides, now that i think about it.

**Comment by lishaduck** [comment]
> I don't see any downsides to it, only upsides, now that i think about it.

If I can get tsup to bundle constants, yeah, that'd work.
Right now, it's not bundling constants, which breaks a ton. I'm going to try both noExternal + devDep and see if that helps.

**Comment by lishaduck** [comment]
> > I don't see any downsides to it, only upsides, now that i think about it.
> 
> If I can get tsup to bundle constants, yeah, that'd work. Right now, it's not bundling constants, which breaks a ton. I'm going to try both noExternal + devDep and see if that helps.

That works! I also fixed the types (more undocumented behavior, sorry 😁)

**Comment by lishaduck** [comment]
Ok, downgraded my node and I can reproduce. I guess it's a node bug? Not blocking prod, just attw.

**Comment by lishaduck** [comment]
> Ok, downgraded my node and I can reproduce. I guess it's a node bug? Not blocking prod, just attw.

Nope, it doesn't support the workspace protocol. Will need to switch to pnpm pack.

**Comment by AndreaPontrandolfo** [comment]
> > Ok, downgraded my node and I can reproduce. I guess it's a node bug? Not blocking prod, just attw.
> 
> Nope, it doesn't support the workspace protocol. Will need to switch to pnpm pack ([arethetypeswrong/arethetypeswrong.github.io#173 (comment)](https://github.com/arethetypeswrong/arethetypeswrong.github.io/issues/173#issue-2367837825); note that the pr that fixed it was reverted)

Ye, it seems like they deprecated the version with the PNPM support because it was breakign everything. Crazy!

File: packages/create-sheriff-config/package.json [hunk 5]
```diff
@@ -26,7 +25,7 @@
     "yargs": "^17.7.2"
   },
   "bin": {
-    "create-sheriff-config": "./dist/index.js"
+    "create-sheriff-config": "dist/index.js"
```
Comment (AndreaPontrandolfo) [inline packages/create-sheriff-config/package.json:28]: why was this needed?

Comment (lishaduck) [reply packages/create-sheriff-config/package.json:28]: It's not, technically...
I ran [`npm pkg fix`](https://docs.npmjs.com/cli/v10/commands/npm-pkg#:~:text=delete%20scripts.build-,npm%20pkg%20fix,-Auto%20corrects%20common) to normalize the `package.json`s. It makes what goes in node_modules the same, otherwise it's different. It's so that the server serves the same thing, so that the (p)npm client(s) can be more forgiving while still using a single format in-memory.
Canonically, `bin`s apparently oughtn't have leading `./`, but (p)npm fixes it while publishing. This just commits those changes.

**Comment by lishaduck** [comment]
Conflicts! Already!
I despair 😫

🤣

**Comment by lishaduck** [comment]
~~This should pass ci now~~ and have 0 merge conflicts! Yay!
Only thought left is that maybe we should add a tsup-config internal package to keep it all in sync.

**Comment by lishaduck** [comment]
Also, it should be noted that despite attw claiming we support node 10, we don't 😅
I just suppressed the node10 errors.

**Comment by lishaduck** [comment]
Forgot to push 🤦‍♂️

**Comment by AndreaPontrandolfo** [comment]
> ~This should pass ci now~ and have 0 merge conflicts! Yay! Only thought left is that maybe we should add a tsup-config internal package to keep it all in sync.

Nah, it's fine. The eslint config and the CLI have different requirements. It makes sense that they rely on different bundling configs.

**Comment by AndreaPontrandolfo** [comment]
CI fails with:

> Error:   1:30  error  Parse errors in imported module 'eslint-config-sheriff': sourceType 'module' is not supported when ecmaVersion < 2015. Consider adding `{ ecmaVersion: 2015 }` to the parser options. (undefined:undefined)  import/no-named-as-default

I'm not exactly sure why this error surfaces here and now.
Anyway, to fix it, i think it's enough to add a `ecmaVersion` here:

```
      files: [supportedFileTypes],
      plugins: { import: fixupPluginRules(pluginImport) },
      rules: importHandPickedRules,
      languageOptions: {
        parserOptions: {
          sourceType: 'module', // required for https://github.com/import-js/eslint-plugin-import/blob/09476d7dac1ab36668283f9626f85e2223652b37/src/rules/no-default-export.js#L16
        },
      },
```

In this file: `packages/eslint-config-sheriff/src/getBaseConfig.ts`

**Comment by lishaduck** [comment]
> CI fails with:
> 
> > Error:   1:30  error  Parse errors in imported module 'eslint-config-sheriff': sourceType 'module' is not supported when ecmaVersion < 2015. Consider adding `{ ecmaVersion: 2015 }` to the parser options. (undefined:undefined)  import/no-named-as-default
> 
> I'm not exactly sure why this error surfaces here and now. Anyway, to fix it, i think it's enough to add a `ecmaVersion` here:
> 
> ```
>       files: [supportedFileTypes],
>       plugins: { import: fixupPluginRules(pluginImport) },
>       rules: importHandPickedRules,
>       languageOptions: {
>         parserOptions: {
>           sourceType: 'module', // required for https://github.com/import-js/eslint-plugin-import/blob/09476d7dac1ab36668283f9626f85e2223652b37/src/rules/no-default-export.js#L16
>         },
>       },
> ```
> 
> In this file: `packages/eslint-config-sheriff/src/getBaseConfig.ts`. We can try `{ ecmaVersion: "latest" }`, if that works. Otherwise we will have to specify a version, like 6, or 2022...

Yeah, I'd fixed it lcoally but, as always forgot to push 😅

**Comment by AndreaPontrandolfo** [comment]
can you add the changeset?

**Review by AndreaPontrandolfo — DISMISSED** [review]

**Review by AndreaPontrandolfo — APPROVED** [review]

**Comment by lishaduck** [comment]
I'm excited to note that this _drastically_ shrunk the size of `dist/index.js`, from 17,002 lines to 1,159.
